src/arch/cuda/hybridAPI/hapi_impl.cpp

```diff
@@ -519,6 +627,171 @@ hapiWorkRequest::hapiWorkRequest() :
 #endif
   }
 
+// Create POSIX shared memory region accessible to all processes on the same host
+// Invoked by PE rank 0 of each process (no locking needed for SMP)
+void shmCreate() {
+  struct stat shm_file_stat;
+
+  // Create the shared memory file
+  CsvAccess(gpu_manager).shm_name = new char[16];
```

Comment:
This string should be longer (24?), it will overflow in the case that we ever have >99 PEs per node.

Reply:
Good point, will update

Reply:
Changed to use `std::string`